uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeaveRequestTest.php

```diff
@@ -129,6 +129,57 @@ public function testUpdatingALeaveRequestShouldNotDuplicateTheLeaveRequestDates(
     $this->assertEquals($toDate->format('Y-m-d'), $dates[1]->date);
   }
 
+  public function testUpdatingALeaveRequestShouldNotThrowOverLappingLeaveRequestExceptionWhenTheDatesOverlap()
```

Comment:
I think this method name is a bit confusing. It gives me the impression that it should not throw the exception whenever it finds an overlapping leave request, even if that leave request is not the one being updated, which is not the case